Framework/API/src/FileProperty.cpp

```diff
@@ -396,5 +398,44 @@ std::string FileProperty::convertExtension(const std::string &filepath) const {
   fullpath.setExtension(ext);
   return fullpath.toString();
 }
+
+/** Expand user variables in file path.
+ *  On Windows and UNIX, ~ is replaced by the user's home directory, if found.
+ *  If the path contains no user variables, or expansion fails, the path is
+ *  returned unchanged, for errors to be dealt with by the calling function.
+ *  Note: this function does not support the "~user/blah" format for a named
+ *  user's home directory - if this is encountered, the filepath is returned
+ *  unchanged.
+ *  @param filepath The path to expand
+ *  @return The expanded path
+ */
+std::string FileProperty::expandUser(const std::string &filepath) const {
+  auto start = filepath.begin();
+
+  if (*start != '~') // No user variables in filepath
```

Comment:
Is `filepath` guaranteed to be non-zero length? If not we should check if `start = filepath.end()` first.